source/adios2/operator/compress/CompressPNG.cpp

```diff
@@ -0,0 +1,220 @@
+/*
+ * Distributed under the OSI-approved Apache License, Version 2.0.  See
+ * accompanying file Copyright.txt for details.
+ *
+ * CompressPNG.cpp
+ *
+ *  Created on: Jun 10, 2019
+ *      Author: William F Godoy godoywf@ornl.gov
+ */
+#include "CompressPNG.h"
+
+#include <cstring> // std::memset
+
+extern "C" {
+#include <png.h>
+}
```

Comment:
My copy of `png.h` has the following comment, line 366:

```cpp
/* Inhibit C++ name-mangling for libpng functions but not for system calls. */
#ifdef __cpluplus
extern "C" {
#endif 
```

So presumably they are doing the right thing here; and there's no need for the `extern "C"` in our code.